File: 6-maximum_of_array/task.ex
defmodule Task do

  def maxmin(values) do
    [value_head|_] = values
    maxmin(values, value_head, value_head)
  end

  defp maxmin([], max, min) do
    [max|[min]]
  end

  defp maxmin([head_value|tail_values], max, min) do
    max = case max > head_value do
      true -> max
      false -> head_value
    end
    min = case min < head_value do
      true -> min
      false -> head_value
    end
    maxmin(tail_values, max, min)
  end

end

File: 6-maximum_of_array/task.erl
-module(task).

-export([maxmin/1]).

maxmin(Values) ->
  [ValueHead|_] = Values,
  maxmin(Values,ValueHead,ValueHead).

maxmin([], Max, Min) ->
  %%lists:append([[Max], [Min]]);
  [Max|[Min]];
maxmin([ValueHead|ValuesTail], Max, Min) ->
  Max2 = case Max > ValueHead of
    true -> Max;
    false -> ValueHead
  end,
  Min2 = case Min < ValueHead of
    true -> Min;
    false -> ValueHead
  end,
  maxmin(ValuesTail, Max2, Min2).
